benchmark/hardware.sh

```diff
@@ -33,20 +29,68 @@ fi
 mkdir -p clickhouse-benchmark-$SCALE
 pushd clickhouse-benchmark-$SCALE
 
-if [[ ! -f clickhouse ]]; then
-    CPU=$(uname -m)
-    if [[ ($CPU == x86_64) || ($CPU == amd64) ]]; then
-        $FASTER_DOWNLOAD "$AMD64_BIN_URL"
-    elif [[ $CPU == aarch64 ]]; then
-        $FASTER_DOWNLOAD "$AARCH64_BIN_URL"
-    elif [[ $CPU == powerpc64le ]]; then
-        $FASTER_DOWNLOAD "$POWERPC64_BIN_URL"
-    else
-        echo "Unsupported CPU type: $CPU"
-        exit 1
+OS=$(uname -s)
+ARCH=$(uname -m)
+
+DIR=
+
+if [ "${OS}" = "Linux" ]
+then
+    if [ "${ARCH}" = "x86_64" ]
+    then
+        DIR="amd64"
+    elif [ "${ARCH}" = "aarch64" ]
+    then
+        DIR="aarch64"
+    elif [ "${ARCH}" = "powerpc64le" ]
+    then
+        DIR="powerpc64le"
+    fi
+elif [ "${OS}" = "FreeBSD" ]
+then
+    if [ "${ARCH}" = "x86_64" ]
+    then
+        DIR="freebsd"
+    elif [ "${ARCH}" = "aarch64" ]
+    then
+        DIR="freebsd-aarch64"
+    elif [ "${ARCH}" = "powerpc64le" ]
+    then
+        DIR="freebsd-powerpc64le"
+    fi
+elif [ "${OS}" = "Darwin" ]
+then
+    if [ "${ARCH}" = "x86_64" ]
+    then
+        DIR="macos"
+    elif [ "${ARCH}" = "aarch64" -o "${ARCH}" = "arm64" ]
+    then
+        DIR="macos-aarch64"
     fi
 fi
 
+if [ -z "${DIR}" ]
+then
+    echo "The '${OS}' operating system with the '${ARCH}' architecture is not supported."
+    exit 1
+fi
+
+URL="https://builds.clickhouse.com/master/${DIR}/clickhouse"
+echo
+echo "Will download ${URL}"
+echo
+curl -O "${URL}" && chmod a+x clickhouse || exit 1
+echo
+echo "Successfully downloaded the ClickHouse binary, you can run it as:
```

Comment:
The user is not intended to run or install this binary.
Let's remove this and subsequent echo.